src/hooks/OnIncompatibleSchemaChange/useSpecificationIncompatibleSchemaSetting.ts

```diff
@@ -0,0 +1,69 @@
+import { modifyDraftSpec } from 'api/draftSpecs';
+import {
+    useEditorStore_persistedDraftId,
+    useEditorStore_queryResponse_draftSpecs,
+    useEditorStore_queryResponse_mutate,
+} from 'components/editor/Store/hooks';
+import { AutoCompleteOption } from 'components/incompatibleSchemaChange/types';
+import { omit } from 'lodash';
+import { useCallback, useMemo } from 'react';
+import { logRocketEvent } from 'services/shared';
+import { CustomEvents } from 'services/types';
+import { Schema } from 'types';
+import { hasLength } from 'utils/misc-utils';
+
+export default function useSpecificationIncompatibleSchemaSetting() {
+    // Draft Editor Store
+    const draftId = useEditorStore_persistedDraftId();
+    const draftSpecs = useEditorStore_queryResponse_draftSpecs();
+    const mutateDraftSpecs = useEditorStore_queryResponse_mutate();
+
+    const draftSpec = useMemo(
+        () =>
+            draftSpecs.length > 0 && draftSpecs[0].spec ? draftSpecs[0] : null,
+        [draftSpecs]
+    );
+
+    const updateOnIncompatibleSchemaChange = useCallback(
+        async (value: AutoCompleteOption['val'] | undefined) => {
+            if (!mutateDraftSpecs || !draftId || !draftSpec) {
+                logRocketEvent(
+                    `${CustomEvents.INCOMPATIBLE_SCHEMA_CHANGE} : missing critical resources to update draft`,
+                    {
+                        draftIdMissing: !draftId,
+                        draftSpecMissing: !draftSpec,
+                        mutateMissing: !mutateDraftSpecs,
+                    }
+                );
+
+                return Promise.resolve();
+            }
+
+            let spec: Schema = draftSpec.spec;
+
+            if (!hasLength(value)) {
+                spec = omit(spec, 'onIncompatibleSchemaChange');
+            } else {
+                spec.onIncompatibleSchemaChange = value;
+            }
```

Comment:
Please move this over to `workflow-utils`. Trying to get all these kind of schema/property update-ers in one spot to make writing a single "tool" for this easier.

Reply:
Since `addOrRemoveSourceCapture` exists in _src/utils/entity-utils.ts_, I defined a similar function for the specification-level `onIncompatibleSchemaChange` property there as well. It is used in this hook and `generateTaskSpec`.